### ISSUE TITLE
Invalid Server URL or external API not enabled

### ISSUE THREAD

**Issue by chart41** [jitsi/jitsi-meet-electron#293]
![invalid_url](https://user-images.githubusercontent.com/44474464/80069962-43883080-8553-11ea-8afc-0a9687836888.png)

Server works fine via browser and with basic authentication. Public domain, let's encript certificate installed. But when i trying to enter server adress (with\without https://), i getting "Invalid Server URL or external API not enabled".
How to fix it?

**Comment by saghul** [comment]
Are you using basic HTTP auth, as in with an `.htaccess` file?

**Comment by chart41** [comment]
no, it's jitsi's authentication with prosody

**Comment by saghul** [comment]
Have you checked this? https://github.com/jitsi/jitsi-meet-electron#using-it-with-your-own-jitsi-meet-installation

**Comment by chart41** [comment]
for which file i should to use this code?
```
location /external_api.js {
    alias /usr/share/jitsi-meet/libs/external_api.min.js;
}
```

**Comment by chart41** [comment]
Yes. This code exixts in site's config file. What can I do else?

**Comment by noahhaon** [comment]
Make sure you have a trailing slash in the URI

**Comment by duanwujie** [comment]
![image](https://user-images.githubusercontent.com/5893575/80295988-16818b00-87aa-11ea-8c76-64024c6f6c32.png)

**Comment by saghul** [comment]
@duanwujie You also need to have a valid TLS certificate, which won’t be the case if you use an IP address.

**Comment by noFAYZ** [comment]
Same issue i did everything with DNS even i can connect to meet.jit.si

**Comment by ctr49** [comment]
I also notice this from time to time with servers that have external API enabled and a valid certs.
To me it looks like the validation happens while typing (with every keystroke) and then either it does not keep up with long URLs or running into some sort of race condition.
In those cases I always had success with copy and pasting the server URL rather than typing it.
(of course only for cases where external API is enabled and the site has a valid cert)

**Comment by faxotherapy** [comment]
Same issue here, but on MacOS. No problem when using the browser. Once, I input my URL (to my own server) in the Electron app, I got the message “Invalid Server URL”. So, is my Firefox browser more clever than the app?

**Comment by pgenderson** [comment]
Is there an explanation regarding why it works (connection to server is successful) without adding any external api js when connecting via Ubuntu but not Windows?

**Comment by pgenderson** [comment]
@saghul, here's the issue I mentioned to you on the community forum. Others seem to be having the same issue.

**Comment by saghul** [comment]
Can you open the dev console and check the logs? it's now enabled on release builds too.

**Comment by pgenderson** [comment]
I'm not exactly sure what you mean by dev console in Windows' context, but main.txt in home\AppData\Roaming\Jitsi Meet\logs just says several times that Update for version 2.3.1 is not available. \AppData\Roaming\Jitsi Meet\Jitsi Meet\logs is empty.

**Comment by pgenderson** [comment]
Now when I paste the URL in, it doesn't show the error and connects fine (probably because the server's on now), but the URL underlines red when the I click the URL input box and the cursor's blinking. When I type the URL (so without pasting), I get the error but the URL stays (and I can connect fine) and doesn't reset back to the default meet.jit.si when I reopen the settings pane.

**Comment by saghul** [comment]
You can open DevTools with the same key combination that you would in Chrome. I think on Windows it's Ctrl+Alt+i

**Comment by saghul** [comment]
For everyone here: I just committed a patch which greatly simplifies server URL verification, since we now bundle external API, so there is no need to check for it. If you folks can test current master and let me know if it works ok it would be ideal. Cheers!

**Comment by saghul** [comment]
This is now released in 2.4.0.
